LiteCore/Query/Translator/Node.hh

```diff
@@ -89,13 +89,15 @@ namespace litecore::qt {
         ParseContext(ParseContext&&) = default;
 
         ParseDelegate&                           delegate;
-        Arena<>&                                 arena;                    // The arena allocator
-        SelectNode* C4NULLABLE                   select{};                 // The enclosing SELECT, if any
-        std::unordered_map<string, AliasedNode*> aliases;                  // All of the sources & named results
-        std::vector<SourceNode*>                 sources;                  // All sources
-        SourceNode* C4NULLABLE                   from{};                   // The main source
-        Collation                                collation;                // Current collation in effect
-        bool                                     collationApplied = true;  // False if no COLLATE node generated
+        Arena<>&                                 arena;                     // The arena allocator
+        SelectNode* C4NULLABLE                   select{};                  // The enclosing SELECT, if any
+        std::unordered_map<string, AliasedNode*> aliases;                   // All of the sources & named results
+        std::vector<SourceNode*>                 sources;                   // All sources
+        SourceNode* C4NULLABLE                   from{};                    // The main source
+        Collation                                collation;                 // Current collation in effect
+        bool                                     collationApplied = true;   // False if no COLLATE node generated
+        bool                                     hasGroupBy       = false;  // Current SELECT include GROUP_BY
```

Comment:
Putting the flag here only works for the top-level query. If there's a nested SELECT expression, the flag will be set in the nested context and the QueryTranslator won't see it.
Instead, the node whose SQL generation depends on `hasGroupBy` should make that an instance variable, and set it when the node is constructed by looking at its parent SELECT statement.

Reply:
yes. good point. Passed hasGroupBy from ctx to the UnnestNode.